src/tactic/interactive.lean

```diff
@@ -1132,5 +1132,52 @@ add_tactic_doc
   decl_names := [`tactic.interactive.generalize'],
   tags       := ["context management"] }
 
+
+/--
+Query the proof goal and print the skeleton of a proof by
+cases.
+
+For example, let us consider the following proof:
+
+```lean
+example {α} (xs ys : list α) (h : xs ~ ys) : true :=
+begin
+  induction h,
+  pretty_cases,
+    -- Try this:
+    -- case list.perm.nil :
+    -- { admit },
+    -- case list.perm.cons : h_x h_l₁ h_l₂ h_a h_ih
+    -- { admit },
+    -- case list.perm.swap : h_x h_y h_l
+    -- { admit },
+    -- case list.perm.trans : h_l₁ h_l₂ h_l₃ h_a h_a_1 h_ih_a h_ih_a_1
+    -- { admit },
+end
+```
+
+The output helps the user layout the cases and rename the
+introduced variables.
+-/
+meta def pretty_cases : tactic unit := retrieve $ do
+gs ← get_goals,
+trace "Try this: ",
+gs.mmap' $ λ g, do
+{ t : list name ← get_tag g,
+  let vs := t.tail,
+  let ⟨vs,ts⟩ := vs.span (λ n, name.last_string n = "_arg"),
+  set_goals [g],
+  ls ← local_context,
+  let m := native.rb_map.of_list $ (ls.map expr.local_uniq_name).zip (ls.map expr.local_pp_name),
+  let vs := vs.map $ λ v, (m.find v.get_prefix).get_or_else `_,
+  let var_decls := string.intercalate " " $ vs.map to_string,
+  trace!"case {ts.head} : {var_decls}\n{{ admit }," }
```

Comment:
What's the tradeoff between `admit` vs `sorry`? I've only really seen the latter used.

Reply:
They are synonyms. Once upon a time `sorry` was for terms and `admit` was for tactics, cribbed from Coq tactic mode, and then we realized that we can make `sorry` be a tactic too. A similar story lies in the renaming of the `note` tactic to `have` and `pose` to `let`; it is making the tactic mode resemble the term mode a bit more, so people have less to learn.

In the years since, `sorry` has been used near-universally. I think it would not be a problem if we removed `admit` entirely.

Reply:
Based on that, should we use `sorry` here instead?

Reply:
Yes sorry would also work